controllers/jobs.js

```diff
@@ -17,11 +26,35 @@ const createJob = async (req, res) => {
 }
 
 const updateJob = async (req, res) => {
-  res.send("update job")
+  const {
+    user: { userId },
+    body: { company, position },
+    params: { id: jobId },
+  } = req
+  if (company === "" || position === "") {
```

Comment:
Do you know a way to write this expression shorter? If so, why did you use this instead? If not, try to find one!!

Reply:
I could do this, what do you think?

```suggestion
  if (!company || !position) {
```

Reply:
Hm better, but can you write it with just a single `!`?

Reply:
I didn't test, but would this work?

```suggestion
  if (!(company || position)) {
```

Reply:
What would happen for this input?

```javascript
const company = "Fancy Corp."
const position = ""

if (!(company || position)) {
  // the input is invalid, so we expect the code to go down this branch
  console.log('invalid input');
}

Reply:
You're close though!

Reply:
I just tested and it does not work.

It tries to update the job in the database and I get the custom error from mongodb.

I do not know how to write with a single `!`

Reply:
You just need this:

```javascript
if (!(company && position)) {...}
```

Reply:
that worked! Very cool!!! :grin:

Reply:
Now this is much more readable because it maps closely to describing the validation requirement in plain english: "if we do not have company and position."

However, you will have a problem because now our validation logic is more general; we don't validate that `company` and `position` are strings in the first place! Send this data to your API and watch it explode:

```json
{
  "company": {
    "oops": "you didn't expect an object here did you? Objects are truthy so it passes the check!"
  },
  "position": "Senior Bug Creator"
}
```

So, the most defensive implementation is probably

```javascript
if (!(company && postiion && typeof company === 'string' && typeof position === 'string')) { ... };
```

But, that's just one last thing to keep in mind.